web/src/components/toolbar/ColorerOption.vue

```diff
@@ -0,0 +1,67 @@
+<script>
+export default {
+  props: {
+    title: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    value: {
+      type: String,
+      required: false,
+      default: null,
+    },
+    disabled: {
+      type: Boolean,
+      required: false,
+    },
+    options: { // {name: string, options: {name: string, color?: string}[]}[]
+      type: Array,
+      required: true,
+    },
+  },
+  computed: {
+    showSelect() {
+      return !this.value || this.options.length > 1;
+    },
+    filterOptions() {
+      const selected = this.options.find(d => d.value === this.value);
+      return selected ? selected.options : [];
+    },
+    hasOptions() {
+      if (this.options.length === 0) {
+        return false;
+      }
+      if (this.options.length === 1 && !this.options[0].value) {
+        return false;
+      }
+      return true;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+div(v-if="hasOptions")
+  v-toolbar.darken-3(color="primary", dark, flat, dense, :card="false")
+    v-toolbar-title
+      slot(name=title) {{title}}
+
+  v-card.mx-3(flat)
+    v-card-actions(style="display: block")
+      v-select.my-0(:value="value", v-if="showSelect",
+          hide-details, :disabled="disabled",
+          @change="$emit('input', $event)",
+          :items="options", item-text="name")
+      .my-0.option(v-for="o in filterOptions", :key="o.name", :title="o.name")
+        v-icon(:color="o.color") {{ $vuetify.icons.square }}
+        | {{ o.name }}
```

Comment:
I've been struggling with this part for a while.   The color legend looks almost exactly like the controls, so it's very easy to confuse between the checkboxes and this, which is non-interactive.  

I wonder if these colorer cards could be combined with the control boxes, since as far as I can tell the information is duplicated.  The only issue is that the color isn't available when the box is unchecked, which I think we could solve by either changing the color of the unchecked box or putting a color dot to the right of the control.  

![Screenshot from 2019-11-11 08-34-36](https://user-images.githubusercontent.com/4214172/68591204-29074280-045e-11ea-9002-db58c23444d6.png)

I know that a future PR adds group selection to these boxes.  I think this could be rolled into the same control as well.

Reply:
so far if you can filter by selection but color by e.g. data source. In these cases the need to be two separate controls. Yes, when both show the same attribute information is redundant.

Reply:
Thanks. This is a hard control, but I see the case you've described.  I wonder if @waxlamp has any ideas about how to either unify or create distinction between the two.

Reply:
Can we move this discussion to an issue? Due to the deadline it's important to get major functionality merged.